dao/patient/etl-patient-dao.js

```diff
@@ -1,309 +1,354 @@
+      var encTypeNames = data[0];
+      var summaryData = data[1];
+      
+      // Map encounter type ids to names.
+      _.each(encTypeNames.result, function(row) {
+        encounterTypeNames[row.encounter_type_id] = row.name;
       });
-
-      db.queryServer_test(queryParts, function(result) {
-        _.each(result.result, function(row) {
-          row.cur_arv_meds = helpers.getARVNames(row.cur_arv_meds);
-          row.arv_first_regimen = helpers.getARVNames(row.arv_first_regimen);
-          row['encounter_type_name'] = encounterTypeNames[row.encounter_type];
-          row['prev_encounter_type_name'] = encounterTypeNames[row.prev_encounter_type_hiv];
-        });
+      
+      // Format & Clean up raw summaries
+      _.each(summaryData.result, function(summary) {
+        summary.cur_arv_meds = helpers.getARVNames(summary.cur_arv_meds);
+        summary.arv_first_regimen = helpers.getARVNames(summary.arv_first_regimen);
+        summary['encounter_type_name'] = encounterTypeNames[summary.encounter_type];
+        summary['prev_encounter_type_name'] = encounterTypeNames[summary.prev_encounter_type_hiv];
+      });
+      
+      // Return when done.
+      return summaryData;
+    });
+    
+    if(_.isFunction(callback)) {
+      promise.then(function(result) {
         callback(result);
+      }).catch(function(err) {
+        callback(err);
       });
-    },
-    getPatientVitals: function getPatientVitals(request, callback) {
-      var uuid = request.params.uuid;
-      var order =  helpers.getSortOrder(request.query.order);
-      // request.query.page;
-      // request.query.pageSize;
-
-      var queryParts = {
-        columns: request.query.fields || "*",
-        table: "etl.flat_vitals",
-        where: ["uuid = ?", uuid],
-        order: order || [{
-          column: 'encounter_datetime',
-          asc: false
-        }],
-        offset: request.query.startIndex,
-        limit: request.query.limit
-      };
+    }
+    
+    return promise;
+  }
+  
+  function getPatientVitals(request, callback) {
+    var uuid = request.params.uuid;
+    var order =  helpers.getSortOrder(request.query.order);
+    // request.query.page;
+    // request.query.pageSize;
 
-      db.queryServer_test(queryParts, function(result) {
+    var queryParts = {
+      columns: request.query.fields || "*",
+      table: "etl.flat_vitals",
+      where: ["uuid = ?", uuid],
+      order: order || [{
+        column: 'encounter_datetime',
+        asc: false
+      }],
+      offset: request.query.startIndex,
+      limit: request.query.limit
+    };
+    
+    // Use promisified function instead
+    var promise = db.queryDb(queryParts);
+    
+    if(_.isFunction(callback)) {
+      promise.then(function(result) {
         callback(result);
+      }).catch(function(err) {
+        callback(err);
       });
+    }
+    
+    // return the promise
+    return promise;
+  }
+  
+  function getClinicalNotes(request, callback) {
```

Comment:
The request object is hapi specific. Look at line 109 for e.g. The person does 'request.params.uuid'. This means the dao layer is tightly coupled to hapi module. The callback in this case is also tightly coupled to hapi. For e.g. assume the results of my query is the number 404. When I do callback(404), hapi will respond to the client saying that the request was a bad one.